File: Data Structures & Algorithms/is-palindrome/submission-0.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()
        left = 0
        right = len(s) - 1

        while left < right:
            if (ord(s[left]) in range(97, 123) or ord(s[left]) in range(48, 58)) and (
                ord(s[right]) in range(97, 123) or ord(s[right]) in range(48, 58)
            ):
                if s[left] != s[right]:
                    return False
            if ord(s[left]) not in range(97, 123) and ord(s[left]) not in range(48, 58):
                left += 1
            elif ord(s[right]) not in range(97, 123) and ord(s[right]) not in range(
                48, 58
            ):
                right -= 1
            else:
                left += 1
                right -= 1
        return True 

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int maxDepth(TreeNode root) {
        if(root==null)
        {
            return 0;
        }
        else if (root.left==null && root.right==null)
        {
            return 1;
        }

        int maxL = maxDepth(root.left);
        int maxR = maxDepth(root.right);

        if(maxL>=maxR)
        {
            return 1+maxL;
        }
        else
        {
            return 1+maxR;
        }
    }
}
